app_backend/github_client.py

```diff
@@ -0,0 +1,274 @@
+            patch = file_info.get("patch", "")
+
+            if patch:
+                # Add file header
+                diff_content.append(f"diff --git a/{filename} b/{filename}")
+                diff_content.append(f"--- a/{filename}")
+                diff_content.append(f"+++ b/{filename}")
+                diff_content.append(patch)
+                diff_content.append("")  # Empty line between files
+
+        result = "\n".join(diff_content)
+        logger.debug(f"GitHub API: Retrieved actual diff content for {len(data.get('files', []))} files")
+        return result
+
+    def get_all_commit_data(self, commit_hash: str, base_branch: str = "main") -> Tuple[str, List[GitCommit], str, str]:
+        """
+        Get all commit data (merge base, commits, file stats, diff) in a single optimized call.
+
+        Args:
+            commit_hash: The target commit hash
+            base_branch: The base branch (default: main)
+
+        Returns:
+            Tuple of (merge_base, commits, file_stats, actual_diff)
+        """
+        logger.debug(f"GitHub API: Getting all commit data for {commit_hash[:8]} from {base_branch}")
+
+        # First get merge base
+        merge_base_data = self._get_compare_data(base_branch, commit_hash)
+        merge_base = merge_base_data["merge_base_commit"]["sha"]
+
+        # Then get commit range data (will be cached if same as merge_base call)
+        commit_data = self._get_compare_data(merge_base, commit_hash)
+
+        # Extract commits
+        commits = []
+        for commit_item in commit_data["commits"]:
+            commits.append(
+                GitCommit(
+                    hash=commit_item["sha"],
+                    message=commit_item["commit"]["message"],
+                    author=commit_item["commit"]["author"]["name"],
+                    date=commit_item["commit"]["author"]["date"][:10],
+                )
+            )
+
+        # Extract file stats
+        files_changed = len(commit_data.get("files", []))
+        total_additions = sum(f.get("additions", 0) for f in commit_data.get("files", []))
+        total_deletions = sum(f.get("deletions", 0) for f in commit_data.get("files", []))
+
+        diff_stats = [f" {files_changed} files changed"]
+        if total_additions > 0:
+            diff_stats.append(f"{total_additions} insertions(+)")
+        if total_deletions > 0:
+            diff_stats.append(f"{total_deletions} deletions(-)")
+
+        file_stats = []
+        for file_info in commit_data.get("files", []):
+            filename = file_info.get("filename", "")
+            additions = file_info.get("additions", 0)
+            deletions = file_info.get("deletions", 0)
+            changes = file_info.get("changes", 0)
+
+            file_line = f" {filename}"
+            if changes > 0:
+                file_line += f" | {changes} "
+                if additions > 0:
+                    file_line += "+" * min(additions, 20)
```

Comment:
Arbitrary undocumented limits huh? Or is it just an attempt to mimic git format?